cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/DataprocRuntimeJobManager.java

```diff
@@ -396,13 +428,20 @@ private boolean isDeleteLifecycleEnabled(String bucketName) {
    */
   private LocalFile uploadCacheableFile(String bucket, String targetFilePath,
                                         LocalFile localFile)
-    throws IOException, StorageException, InterruptedException {
+    throws IOException, StorageException {
     Storage storage = getStorageClient();
     BlobId blobId = BlobId.of(bucket, targetFilePath);
     Blob blob = storage.get(blobId);
     LocalFile result;
 
     if (blob != null && blob.exists()) {
+
+      if (artifactsCacheablePerCDAPVersion.contains(localFile.getName()) &&
+        // if the version is SNAPSHOT, check if the GCS object creation time is older than the build time.
+        (cdapVersionInfo.isSnapshot() && blob.getCreateTime() < cdapVersionInfo.getBuildTime())) {
```

Comment:
Had a discussion with @masoud-io , I think we can skip this `isSnapshot` check, but always check the creationTime against the buildTime, so that it works independent of snapshot or not.

Reply:
Also, to simplify cleanup (since there could be many GCS objects created over time), we can just set a custom time to `buildTime() + x`, where x is `1 week` if snapshot, otherwise to `2 years` if non-snapshot.

Reply:
ack